content/2016.12.how-to-develop-a-webapp-with-Kinto.md

```diff
@@ -0,0 +1,235 @@
+Title: How to develop a WebApp with Kinto
+Date: 2016-12-01 10:20
+Slug: develop-a-chat-webapp-with-kinto
+Authors: Enguerran Colson
+Summary: An example of how Kinto handles a webapp's data storage.
+
+## Our goal
+
+Hi folks, today, we are gonna create a simple WebApp that allows anyone to chat in a room accessible through a URL.
+
+![](https://d2mxuefqeaa7sj.cloudfront.net/s_4340C716162DF03B2B97575BFA438C542185694D96D86DB6EB44B989957A3ADE_1480537293391_file.png)
+{.image}
+
+* A really simple chat webapp with minimal UI.
+* We will need a bit of JavaScript to enhance and animate interactions within the application and we will also need a way to store and share the app's data. Our front-end should be able to consume an API that would look like `https://api.discute.re/v1/rooms/4815162342/messages` with [discute.re](http://discute.re) as the domain of our application (*discutere* means to chat in Latin).
+
+Our API endpoints could be:
+
+* `/rooms` to return all chat rooms,
+* `/rooms/:id/` to return a particular chat room identified by an ID,
+* `/rooms/:id/messages` to return all messages from a particular room.
+
+To post a message, our front-end will do a POST request to `/rooms/:id/messages`.
+
+We will store each message like this:
+
+    {
+        "message": "I know nothing",
+        "last_modified": 1480340246147,
+        "author": "Jon Snow",
+        "id": "a1e6e879-5c94-4a01-9ec2-28e152d1d8f4"
+    }
+
+
+The JavaScript code could use whatever library or latest hype JS framework but in our example we will stick with modern [Vanilla JS](http://vanilla-js.com/) and use top notch [ES2015 syntax](https://babeljs.io/docs/learn-es2015/).
+
+## API creation
+
+Hmm… what backend should we use? Django? Express? Firebase?… oh wait, why don’t we give [Kinto](http://www.kinto-storage.org/) a try?
+
+Kinto is a perfect fit for this application as once installed, Kinto provides a flexible enough API tohandle other APIs.
```

Comment:
- *to handle* (missing space)
- I'd rephrase saying *Kinto provides a flexible & generic enough API to address the needs of many different kinds of  application*